tests/sundials/kinsol_06.cc

```diff
@@ -14,7 +14,6 @@
 
 #include <deal.II/base/parameter_handler.h>
 
-#include <deal.II/lac/full_matrix.h>
```

Comment:
While there, also remove this unnecessary inclusion in the test from which the new testcase is adapted.